modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotMetadata.java

```diff
@@ -97,6 +101,7 @@ public SnapshotMetadata(
         List<Integer> grpIds,
         Set<String> bltNodes,
         Set<GroupPartitionId> pairs,
+        @Nullable Long lastSegIdx,
```

Comment:
It seems `lastSegIdx` can be null only when we read existing snapshot metadata from disk.

Reply:
It could be null if wal() is disabled by user while snapshot is creating.